File: 06-03-paging/java-06-03-paging/app/src/main/java/com/example/paging/PokeAPI.java
package com.example.paging;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface PokeAPI {
    @GET("pokemon/")
    Call<Response> listPokemons();

    @GET("pokemon/")
    Call<Response> listPokemons(@Query("offset") String offset, @Query("limit") String limit);
}

File: 08-02-network-image-coroutine/app/src/main/java/com/example/viewModelImageCoroutine/PokeAPI.kt
package com.example.viewModelImageCoroutine

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface PokeAPI {
    @GET("pokemon/")
    fun listPokemons(): Call<Response>

    @GET("pokemon/")
    fun listPokemons(@Query("offset") offset: String, @Query("limit") limit: String): Call<Response>

    @GET("pokemon/{pid}/")
    suspend fun getPokemon(@Path("pid") pid: Int): PokemonResponse
}
